pkg/auth/service.go

```diff
@@ -978,6 +981,34 @@ func (s *DBAuthService) Authorize(ctx context.Context, req *AuthorizationRequest
 	return &AuthorizationResponse{Allowed: true}, nil
 }
 
+func (s *DBAuthService) ClaimTokenIDOnce(ctx context.Context, tokenID string, expiresAt int64) (bool, error) {
```

Comment:
Can you add validation checks for tokenID and expiresAt.
- tokenID not empty
- expiresAt > 0